src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/StringCollectionTextWriter.cs

```diff
@@ -41,12 +42,12 @@ public override Encoding Encoding
         /// A collection of entries buffered by this instance of <see cref="StringCollectionTextWriter"/>.
         /// </summary>
         // internal for testing purposes.
-        internal BufferEntryCollection Buffer { get; }
+        internal BufferedHtmlContent Content { get; }
 
         /// <inheritdoc />
         public override void Write(char value)
         {
-            Buffer.Add(value.ToString());
+            Content.Append(value.ToString());
```

Comment:
Ah, for some reason I thought you had a char overload. Anyway, did you consider storing more types (char[], char) "natively" in the Entries array so that you can write them to the textwriter without creating string first?